recirq/qcqmc/analysis.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2024 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Mapping, Tuple, Union
+
+import attrs
+import numpy as np
+
+from recirq.qcqmc import (
+    blueprint,
+    config,
+    data,
+    experiment,
+    hamiltonian,
+    shadow_tomography,
+    trial_wf,
+)
+
+
+@attrs.frozen(repr=False)
+class OverlapAnalysisParams(data.Params):
+    """Class for storing the parameters that specify an OverlapAnalysisData.
+
+    This stage of pipeline is where the data from a (real or simulated) experiment is
+    analyzed and we reconstruct the trial wavefunction from the classical shadow.
+
+    Args:
+        name: A `Params` name for this experiment.
+        experiment_params: Backreference to the `ExperimentParams` preceding this stage.
+            We should have experiment_params.blueprint_params be a blueprint for an experiment
+            that does shadow tomography on a trial wavefunction.
+        k_to_calculate: For shadow tomography, we use a "median of means" approach to statistical
+            robustness as explained in Huang et. al. (https://arxiv.org/abs/2002.08953). These
+            k's serve as keys for the groups.
+    """
+
+    name: str
+    experiment_params: experiment.SimulatedExperimentParams
+    k_to_calculate: Tuple[int, ...] = attrs.field(converter=tuple)
+    path_prefix: str = ""
+
+    def __attrs_post_init__(self):
+        if not isinstance(
+            self.experiment_params.blueprint_params, blueprint.BlueprintParamsTrialWf
+        ):
+            raise ValueError(
+                "Experiment must be build from a blueprint derived from a BlueprintParamsTrialWf."
+            )
+
+    @property
+    def path_string(self) -> str:
+        return self.path_prefix + config.OUTDIRS.DEFAULT_ANALYSIS_DIRECTORY + self.name
+
+    def _json_dict_(self):
+        return attrs.asdict(self)
+
+
+@attrs.frozen(eq=False)
+class OverlapAnalysisData(data.Data):
+    """Container for analysis of overlap.
```

Comment:
This is a pretty short doc string for this class.  It basically just restates the name.

Reply:
Fixed.